src/main/kotlin/com/criptext/mail/push/data/PushAPIRequestHandler.kt

```diff
@@ -75,9 +77,23 @@ class PushAPIRequestHandler(private val not: CriptextNotification,
                 .flatMap { Result.of { emailDao.toggleCheckingRead(listOf(it.id), false) } }
                 .flatMap { Result.of { apiClient.postOpenEvent(listOf(metadataKey)) } }
         when(operation){
-            is Result.Success -> manager.cancel(notificationId)
+            is Result.Success -> {
+                val notCount = storage.getInt(KeyValueStorage.StringKey.NewMailNotificationCount, 0)
+                manager.cancel(notificationId)
+                if((notCount - 1) == 0) {
+                    manager.cancel(CriptextNotification.INBOX_ID)
+                }else{
+                    storage.putInt(KeyValueStorage.StringKey.NewMailNotificationCount, notCount - 1)
+                }
+            }
```

Comment:
This is too repetitive. You need a beter abstraction that cancels the notification and updates the counter